lib/run-context.js

```diff
@@ -119,7 +119,9 @@ RunContext.prototype._run = function () {
  */
 RunContext.prototype.toPromise = function () {
   return new Promise(function (resolve, reject) {
-    this.on('end', resolve);
+    this.on('end', function () {
+      resolve(this.targetDirectory);
```

Comment:
Actually, the `listener` functions of Node `EventEmitter`s are bound to the same EventEmitter, so this works here